### PR TITLE
Fix a bug in skiplayernorm fusion pattern 2

### PR DESCRIPTION
**Pull request by yufenglee**
For skiplayernorm fusion pattern 2, its input[0] should be equal to the input[0] of Add_1, but is overridden by the input[0] of Add_2.

### CODE REVIEW

File: onnxruntime/core/optimizer/skip_layer_norm_fusion.cc [hunk 1]
```diff
@@ -57,13 +57,18 @@ static bool CheckFirstAdd(Node& add, ProviderType providertype) {
...
-static bool CheckSecondAdd(Node& add, ProviderType providertype) {
+static bool CheckSecondAdd(Graph& graph, Node& add, ProviderType providertype) {
   if (providertype != add.GetExecutionProviderType() ||
       !IsSupportedDataType(add) ||
       add.GetOutputEdgesCount() != 1) {
     return false;
   }
 
+  // The 2nd input should be a constant value
+  if (!graph_utils::NodeArgIsConstant(graph, *(add.MutableInputDefs()[1]))) {
```
Comment (tianleiwu) [inline onnxruntime/core/optimizer/skip_layer_norm_fusion.cc:68]: Do we need check the input shape for this?

Comment (yufenglee) [reply onnxruntime/core/optimizer/skip_layer_norm_fusion.cc:68]: yes, just under the constant check

File: onnxruntime/core/optimizer/skip_layer_norm_fusion.cc [hunk 2]
```diff
@@ -230,18 +234,18 @@ Status SkipLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
                                                "SkipLayerNormalization",
                                                "fused SkipLayerNorm subgraphs ",
                                                skip_layer_norm_input_defs,
-                                               {}, {}, kMSDomain);
+                                               ln_node.MutableOutputDefs(), {}, kMSDomain);
```
Comment (tianleiwu) [inline onnxruntime/core/optimizer/skip_layer_norm_fusion.cc:237]: Why not use graph_utils::MoveAllNodeOutputs to move outputs (including edges)?

Comment (yufenglee) [reply onnxruntime/core/optimizer/skip_layer_norm_fusion.cc:237]: Actually MoveAllNodeOutputs is useless. All the edge will be re-built in graph.resolve

**Pull request merged**